# **Final Project** :- **COVID-19 Scandinavia**

> Report by Shubham Ingole *(s200092)*, Emily Grigg *(s196492)* and Thakshina Tharmapalan *(s196628)* | *Updated on 15 May 2020*

**DTU - Technical University of Denmark**

**Course:** 02806 Social Data Analysis and Visualization

**Guide:** Sune Lehmann Jørgensen

**Project details:** Final Course Project, Assignment B - submitted on May 15, 20199


> # **Table of Content**

1. **Motivation**
2. **Basic Stats**
3. **Data Analysis**
4. **Genre**
5. **Visualisation**
6. **Discussion**
7. **Contribution**
8. **Refernece**





In [0]:
##  NOTE: Run this code so the notebook works ##
!pip install wget
!pip install chart-studio

#**1. Motivation** 
Around half of the world's population is on lockdown in an attempt to stop the spread of COVID-19, a public health emergency that has claimed thousands of lives and sparked fears of the worst global recession since the great Depression. In the final project of social data analysis and visualization, we have the opportunity to explore and analyze the current COVID situation. Our approach to the project work is challenging ourselfs and showcase something that the user can quickly observe and take insights. Although, there are numerous live visuals counts and dashboard to get an insight into the impact of the pandemic, we desire to do something specific visualization for the audience. We want to address the impact and response of the countries which are similar in geographically, population, culture. Henceforth, we decided to analyze the current situation in Scandinavia.
 
To execute the formation of Part-B of the project, we have made a video for the introduction of the topic including implementation plan and few stasticts about the projecct. Here is a link to our video explanation of Project-A of the project: [**Cronavisrus In Scandinavia**](https://youtu.be/IyYrRpZ0l9M)

For a user it is always annonying to see long code before the actual infographics hencce for better visual experience we also have created a website which is also a part of the Project-B which can be seen here : [**Coronavirus In Scandinavia**](http://guard.bigredbands.org/covid.html)



**What is the dataset?**  
In this project we be will performing exploratory data analysis of the Novel Corona Virus 2019 Dataset. Our data set is about the Covid-19 pandemic also known as coronavirus. The data set we referred from several different sources that concludes instutuions and private orginisations which includes , [Our world in data](https://ourworldindata.org/), JHU CSSE institute, [Statista](https://www.statista.com/The), [FHI](https://www.fhi.no/sv/smittsomme-sykdommer/corona/dags--og-ukerapporter/dags--og-ukerapporter-om-koronavirus/), . The data set has been altered in regard to the requried data analysis and has been saved [*here*](https://github.com/shubhamrmcf1295/misc). This has daily information on the number of cases, deaths, and recovery in 2019-2020. It contains a set of files for, ‘covid_19_data’ for Scandinavia, contains the following data about:

* Observation Date - Date of the observation in YYYY/MM/DD

* Location- Country of observation

* Confirmed - Cumulative number of confirmed cases 

* Deaths - Cumulative number of deaths 

* Recovered - Cumulative number of recovered cases

* Active cases - Cumulative active postive cases

* Testing – Cumulative number of testing done by country

* Age/gender – Number of infected Age group and biological sex

* Population - Population density of a counrty 











**Why were these datasets chosen?** 

The **COVID-19** pandemic, also famous as the coronavirus pandemic now a days, is an ongoing epidemic of coronavirus disease 2019 (COVID 19). The outbreak was identified in Wuhan, China, in December 2019. And within a month the ***“World Health Organization”*** declared a Public Health Emergency of International Concern on 30 January, and a pandemic on 11 March.  The virus is primarily spread between people in close contact, often via small droplets produced by coughing, sneezing, and talking. As of now, more than 3 million cases of COVID-19 have been reported in over 187 countries and territories, resulting in more than 274,000 deaths. More than 1.31 million people have recovered[(1)](https://www.who.int/).

As the corona virus spreads globally, we’re witnessing a number of national responses to the crisis. We want to take a closer look at the response to the pandemic by countries that are similar geographically, culturally and population wise, but also when it comes to the healthcare system. **This review will address analysis about the response of COVID-19 by Scandinavia that includes the countries of Denmark, Sweden, Finland, Iceland and Norway.**  Moreover, as all other the countries, they have implemented different measures to slow the spread of the virus. This ranges from mandatory geographic quarantines to non-mandatory recommendations to stay at home. 

***Response to the pandemic:***

It is interesting to see how Scandinavian countries manage to contain the virus spread. It is relevant to roughly look into the different strategies of the Scandinavian countries. The populations of **Denmark, Norway** and **Finland** are similar, and they have chosen very similar strategies after their first cases at the end of February. The first case of COVID-19 was reported in **Denmark** and **Norway** in the late February. Within the next two weeks, countries announces a national lockdown also imposing a travel ban and closing the borders. However, Finland had noticed the first case of corona at the end of January, but managed to contain the spread at the time by quarantine and contact tracing. A month later, on February 26, Finland identified another case of corona and the spread. Hence, Finland declared a state of emergency on March 16, closed schools and public facilities and restrcited the national borders[(1)](https://www.diis.dk/en/research/the-nordics-and-covid-19-together-or-apart).

On the other hand, **Sweden** with roughly double inhabitants of the other countries, has chosen a very different strategy. They have maintained society almost the same way as before its first case in late January. They have borders open for other members of the European Union, and schools, restaurants, shops and bars remain open. **Iceland** has a small population with around 365 000 people. However, it also has an infection rate of 1 case per 204 inhabitants, one of the highest in world. This could be attributed to more tests carried out per capita than any other country[(1)](https://www.diis.dk/en/research/the-nordics-and-covid-19-together-or-apart).

**What was the goal for the end user's experience?**   

Though the other mediums were in the front row, constantly supplying viewers with vital information and helping them keep up-to-date with all the topics related to this case, our goals are considerably different, as illustrated below.
> * Gathering relevant and updated data and processing it get aware about the current **Covid-19 epidemic**.
* Illustrating an outline of **Covid-19 impact** by addressing the infected confirmed cases, deaths, and recoveries over time.
* Highlighting the most affected people with respect to **age group and gender.**
* The impact of the **virus spread** in regard to the **population density** of Scandinavian countries with the help of geo maps.
* Comparison between the Scandinavian and their responses, by showcasing the trend about the **testing** of the infected person which is performed over time to contain the virus.



Each different strategy is not static, and will change in different times in the different countries. In other words, there is no specific conclusion regarding the outbreak or the cure about virus from the study. There is no known "best approach." However, this project aims to give the reader a deeper presentation of the data of COVID-19 dataset. This, in addition to knowledge about Scandinavian resposes, are visualized in interactive way to extend the significance of the illustrations to the user.  


#**2. Basic Stats**
**Choices in data cleaning and preprocessing**  
The key factor behind the collection of data is that data almost never arrives in a shape that is ready for us. For our data cleaning, we decided to focus on the Scandinavian countries. Therefore, we eliminate the countries from the dataset other than **Denmark, Finland, Iceland, Norway, and Sweden.** 



**Data for covid impact.**
> **Confirmed, deaths and recovered** are in different CSV files. that makes it difficult for plotting them in the same data visualization.

>**Confirmed, deaths and recovered** are not exactly matched because of missing data, and some countries don’t report Province/State-wise stat for recovered.

> Data for **age group** is in different csv and not in the same format, whihc has been aligned and merged 

In [0]:
#Imports
import pandas as pd
import plotly.io as pio
import plotly.express as px
import json
import io
import requests
import wget
from pandas.io.json import json_normalize #package for flattening json in pandas df
import matplotlib.pyplot as plt
import plotly.graph_objects as go
import os
import PIL
import plotly as py
import numpy as np
import tempfile


In [0]:
data = pd.read_csv('https://raw.githubusercontent.com/CSSEGISandData/COVID-19/master/csse_covid_19_data/csse_covid_19_time_series/time_series_covid19_confirmed_global.csv')
data.head()

Province/State Country/Region      Lat  ...  5/10/20  5/11/20  5/12/20
0            NaN    Afghanistan  33.0000  ...     4402     4687     4963
1            NaN        Albania  41.1533  ...      868      872      876
2            NaN        Algeria  28.0339  ...     5723     5891     6067
3            NaN        Andorra  42.5063  ...      755      755      758
4            NaN         Angola -11.2027  ...       45       45       45

[5 rows x 116 columns]

The data shows the number of confirmed cases each day for every country in the world that has any cases of Covid-19. The data begins at 1/22/20 and updates dynamically everyday, and includes the data of yesterday. That means it shows the total of confirmed cases for a day, that is the sum of confirmed cases from the day before and the new cases from that particular day. We now filter out the data we want, that is the data for the Scandinavian countries; Denmark, Finland, Iceland, Norway and Sweden.


In [0]:
#Making a vector of the Scandinavian countries in order to clean the data with relevant statistics 
my_countries = ['Denmark', 'Finland', 'Iceland', 'Norway', 'Sweden']
#code of the countries are needed for plotting the statistic accordingly into a worldmap
code = ['DNK', 'FIN', 'ISL', 'NOR', 'SWE' ]
my_data = data.loc[data['Country/Region'].isin(my_countries)]

In [0]:
my_data

Province/State Country/Region      Lat  ...  5/10/20  5/11/20  5/12/20
92   Faroe Islands        Denmark  61.8926  ...      187      187      187
93       Greenland        Denmark  71.7069  ...       11       11       11
94             NaN        Denmark  56.2639  ...    10429    10513    10591
106            NaN        Finland  64.0000  ...     5962     5984     6003
130            NaN        Iceland  64.9631  ...     1801     1801     1801
175            NaN         Norway  60.4720  ...     8105     8132     8157
205            NaN         Sweden  63.0000  ...    26322    26670    27272

[7 rows x 116 columns]

The data shows that Faroe Islands and Greenland is a province of Denmark, and we use groupby('Country/Region).sum() in order to sum up the cases for both provinces and add them to the confirmed cases of Denmark. The number of cases in both Faroe Islands and Greenland has been quite stable since 4/4/20, the latter has not had any new cases as of 5/12/20, and in Faroe Islands the increase is 3 cases since. When looking at 5/12/20 the total of confirmed cases of total cases in both provinses is (187+11)/10591 = 0.00187, which is a very small percentage. This means the cases in the provinces does not have a major impact on the cases in Denmark, however we have chosen to include them in Denmark.

The following table shows the last 14 days of the desired data, include the a column of the code for the respective countries in order to get in in a worldmap in a plotly graph. 

In [0]:
#Group the data by the scandinavian country 
my_data = my_data.groupby('Country/Region').sum()
#Drop Lat and Lon coulmns 
my_data = my_data.drop(columns = ['Lat', 'Long'])
#Add one column with the respective code for the respective countries in order to get it in a worldmap
my_data['Code'] = code 
#my_data with respective countrycode with shows the last 15 columns
my_data.iloc[:,-15:] 

4/29/20  4/30/20  5/1/20  ...  5/11/20  5/12/20  Code
Country/Region                            ...                        
Denmark            9206     9356    9509  ...    10711    10789   DNK
Finland            4906     4995    5051  ...     5984     6003   FIN
Iceland            1797     1797    1798  ...     1801     1801   ISL
Norway             7710     7738    7783  ...     8132     8157   NOR
Sweden            20302    21092   21520  ...    26670    27272   SWE

[5 rows x 15 columns]

The map shows that Sweden has highest number of cases today...  

**Data for age group**

We collected data on age and gender distributions for each of the five countries. 

In [0]:
import pandas as pd
import plotly.io as pio
import plotly.express as px
import json
import io
import requests
import matplotlib.pyplot as plt

#Getting Nordic Data
s=requests.get("https://raw.githubusercontent.com/eng22/misc/master/sweden_region.tsv").content
sw_region = pd.read_csv(io.StringIO(s.decode('utf-8')), delimiter="\t")

s=requests.get("https://raw.githubusercontent.com/eng22/misc/master/sweden_new.tsv").content
sw_new = pd.read_csv(io.StringIO(s.decode('utf-8')), delimiter="\t")

s=requests.get("https://raw.githubusercontent.com/eng22/misc/master/sweden_gender.tsv").content
sw_gender = pd.read_csv(io.StringIO(s.decode('utf-8')), delimiter="\t")

s=requests.get("https://raw.githubusercontent.com/eng22/misc/master/sweden_cdeaths.tsv").content
sw_cdeaths = pd.read_csv(io.StringIO(s.decode('utf-8')), delimiter="\t")

s=requests.get("https://raw.githubusercontent.com/eng22/misc/master/sweden_age.tsv").content
sw_age = pd.read_csv(io.StringIO(s.decode('utf-8')), delimiter="\t")

s=requests.get("https://raw.githubusercontent.com/eng22/misc/master/norway_regions.csv").content
nw_region = pd.read_csv(io.StringIO(s.decode('utf-8')))

s=requests.get("https://raw.githubusercontent.com/eng22/misc/master/norway_origin.csv").content
nw_origin = pd.read_csv(io.StringIO(s.decode('utf-8')))

s=requests.get("https://raw.githubusercontent.com/eng22/misc/master/norway_new.csv").content
nw_new = pd.read_csv(io.StringIO(s.decode('utf-8')))

nw_gender = pd.read_csv('https://raw.githubusercontent.com/eng22/misc/master/nw_age_gender.csv?fbclid=IwAR0BzZVUKHdmifHaFRtC6wTYMwhONrzYM0wk2DxXotp2P9eiKrE3h4SVFZU')
s=requests.get("https://raw.githubusercontent.com/eng22/misc/master/norway_cdeaths.csv").content
nw_cdeaths = pd.read_csv(io.StringIO(s.decode('utf-8')))

s=requests.get("https://raw.githubusercontent.com/eng22/misc/master/norway_age.csv").content
nw_age = pd.read_csv(io.StringIO(s.decode('utf-8')))

s=requests.get("https://raw.githubusercontent.com/eng22/misc/master/iceland_region.csv").content
ic_region = pd.read_csv(io.StringIO(s.decode('utf-8')))

s=requests.get("https://raw.githubusercontent.com/eng22/misc/master/iceland_new.csv").content
ic_new = pd.read_csv(io.StringIO(s.decode('utf-8')))
s=requests.get("https://raw.githubusercontent.com/eng22/misc/master/iceland_gender.csv").content
ic_gender = pd.read_csv(io.StringIO(s.decode('utf-8')))

s=requests.get("https://raw.githubusercontent.com/eng22/misc/master/iceland_age.csv").content
ic_age = pd.read_csv(io.StringIO(s.decode('utf-8')))
ic_age.columns = ['Age', 'NumberOfCases']
ic_age.loc[-1] = ['1 year of younger', '2'] 
ic_age.index = ic_age.index + 1
ic_age = ic_age.sort_index() 

s=requests.get("https://raw.githubusercontent.com/eng22/misc/master/finland_region.csv").content
fn_region = pd.read_csv(io.StringIO(s.decode('utf-8')))

s=requests.get("https://raw.githubusercontent.com/eng22/misc/master/finland_new.tsv").content
fn_new = pd.read_csv(io.StringIO(s.decode('utf-8')), delimiter="\t")

s=requests.get("https://raw.githubusercontent.com/eng22/misc/master/finland_gender.csv").content
fn_gender = pd.read_csv(io.StringIO(s.decode('utf-8')), delimiter="\t")

s=requests.get("https://raw.githubusercontent.com/eng22/misc/master/finland_cdeaths.tsv").content
fn_cdeaths = pd.read_csv(io.StringIO(s.decode('utf-8')), delimiter="\t")
 
s=requests.get("https://raw.githubusercontent.com/eng22/misc/master/finland_age.tsv").content
fn_age = pd.read_csv(io.StringIO(s.decode('utf-8')), delimiter="\t")
fn_age.columns = ['Age', 'NumberOfCases']
fn_age.loc[-1] = ['0-9 years', '108'] 
fn_age.index = fn_age.index + 1
fn_age = fn_age.sort_index() 


s=requests.get("https://raw.githubusercontent.com/eng22/misc/master/denmark_region.tsv").content
dk_region = pd.read_csv(io.StringIO(s.decode('utf-8')), delimiter="\t")

s=requests.get("https://raw.githubusercontent.com/eng22/misc/master/denmark_new.tsv").content
dk_new = pd.read_csv(io.StringIO(s.decode('utf-8')), delimiter="\t")

s=requests.get("https://raw.githubusercontent.com/eng22/misc/master/denmark_cdeaths.tsv").content
dk_cdeaths = pd.read_csv(io.StringIO(s.decode('utf-8')), delimiter="\t")

s=requests.get("https://raw.githubusercontent.com/eng22/misc/master/denmark_age_gender.tsv").content
dk_age_gender = pd.read_csv(io.StringIO(s.decode('utf-8')), delimiter="\t")
sum_gender = dk_age_gender['Women'] + dk_age_gender['Men']
dk_age_gender["NumberOfCases"] = sum_gender

The following data is how most of the countries has categories number of cases in certain agegroup, showing data for Sweden. Sweden has 7 cases that are unknown, we chose to remove that row in order of simplification as the other data has no unknown cases at the present situation. In the data prosessing, we clean the data in same format, and rename the column 'NumberOfCases' to the respective country.

In [0]:
sw_age

Age  NumberOfCases
0   9 years and younger            101
1           10-19 years            233
2           20-29 years           1425
3           30-39 years           1884
4           40-49 years           2469
5           50-59 years           3314
6           60-69 years           2507
7           70-79 years           2462
8           80-90 years           2955
9    90 years and older           1569
10              Unknown              7

In [0]:
sw_age_bar = sw_age
sw_age_bar.columns = ['Age', 'Sweden'] #Change NumberOfCases name to Sweden
sw_age_bar = sw_age_bar[:-3] #Remove unknown, 80-89 years and 90 and older and put it in common column
sw_age_bar  = sw_age_bar.append({'Age' : '80 years and older' , 'Sweden' : 4524} , ignore_index=True)
#common column for 80-89 and 90 and older to 80 and older 
#####################################
nw_age_bar = nw_age
nw_age_bar.columns = ['Age', 'Norway']
nw_age_bar = nw_age[:-2] #drop last two columns and make it common for above 80 years
nw_age_bar = nw_age_bar.append({'Age' : '80 years and older' , 'Norway' : 459} , ignore_index=True)
######################################
fn_age_bar = fn_age
fn_age_bar.columns = ['Age', 'Finland']
#######################################
dk_age_bar = dk_age_gender
dk_age_bar = dk_age_gender.drop(columns =['Women', 'Men']) #remove gender to make it similar to the others
dk_age_bar.columns = ['Age', 'Denmark']
dk_age_bar = dk_age_bar.drop(dk_age_bar.index[-1])
dk_age_bar = dk_age_bar.drop(dk_age_bar.index[-1])
dk_age_bar = dk_age_bar.append({'Age' : '80 years and older' , 'Denmark' : 845} , ignore_index=True)

So far the data for all countries except Iceland has been modified as the same format as in the data shown for Sweden age distribution. However, the age has classified differently as shown below. 

In [0]:
ic_age_bar = ic_age
ic_age_bar.columns = ['Age', 'Iceland']
ic_age_bar

Age  Iceland
0     1 year of younger        2
1             1-5 years        4
2            6-12 years        8
3           13-17 years        8
4           18-29 years       43
5           30-39 years       35
6           40-49 years       41
7           50-59 years       43
8           60-69 years       32
9           70-79 years        8
10          80-89 years       10
11          90-99 years        2
12  100 years and older        1

Classification is different than the others, that is for the younger age under 30 years, which makes it different to compare the younger age group with according to the other countries. However, the normalization shows that the first three age groups represent data that is significantly smaller that the agegroup above 18 years. The minor group sums up to a toal of rougly 5% of the data. For simplification, we choose to classify agegroup of '1 year of younger', '1-5 years' and '6-12 years' into one and only age group: '9 years and younger'. Furthermore, we observe that the agegroup for '13-17 years' consists of around 4 % of the data, that can be classified as in age group '10-19 years' and '18-29 years' to '20-29' years. This means the data for Iceland does not represent the true observed values for agegroup below 30. Covid-19 data has unknown amount of dark numbers and the data is very dependent on the test capacity for each countries. 

In [0]:
#same for iceland 
ic_age_bar['Iceland'] = [int(i) for i in ic_age_bar['Iceland']]
######################
#drop last 3 columns 
ic_age_bar = ic_age_bar[:-3]
ic_age_bar  = ic_age_bar.append({'Age' : '80 years and older' , 'Iceland' : 13} , ignore_index=True)
ic_age_bar['Iceland'] = ic_age_bar['Iceland']/ ic_age_bar['Iceland'].sum()

In [0]:
ic_age_bar

Age   Iceland
0    1 year of younger  0.008439
1            1-5 years  0.016878
2           6-12 years  0.033755
3          13-17 years  0.033755
4          18-29 years  0.181435
5          30-39 years  0.147679
6          40-49 years  0.172996
7          50-59 years  0.181435
8          60-69 years  0.135021
9          70-79 years  0.033755
10  80 years and older  0.054852

In [0]:
age1 = ic_age_bar[0:3].sum()
age1

Age        1 year of younger1-5 years6-12 years
Iceland                               0.0590717
dtype: object

In [0]:
ic_data = pd.DataFrame({'Iceland':[0.0590717,0.033755, 0.181435, 0.147679, 0.172996, 0.181435, 0.135021, 0.033755, 0.054852]})
ic_age_bar = dk_age_bar
ic_age_bar['Iceland'] = ic_data
ic_age_bar = ic_age_bar.drop(columns =['Denmark'])
ic_age_bar

Age   Iceland
0  9 years and younger  0.059072
1          10-19 years  0.033755
2          20-29 years  0.181435
3          30-39 years  0.147679
4          40-49 years  0.172996
5          50-59 years  0.181435
6          60-69 years  0.135021
7          70-79 years  0.033755
8   80 years and older  0.054852

In [0]:
#Now placing the data in same dataframe
bar_stack = sw_age_bar
bar_stack['Norway'] = nw_age_bar['Norway']/nw_age_bar['Norway'].sum()
bar_stack['Denmark'] = dk_age_bar['Denmark']/dk_age_bar['Denmark'].sum()
#Cannot sum data for Finland because they are stated as strings, so convert the list of string to list of int.
bar_stack['Finland'] = [int(i) for i in fn_age_bar['Finland']]
bar_stack['Finland'] = bar_stack['Finland']/ bar_stack['Finland'].sum()
bar_stack['Sweden'] = bar_stack['Sweden']/ bar_stack['Sweden'].sum()
bar_stack['Iceland'] = ic_age_bar['Iceland']
bar_stack = bar_stack.round(3)  #NEW 
bar_stack

Age  Sweden  Norway  Denmark  Finland  Iceland
0  9 years and younger   0.005   0.013    0.013    0.023    0.059
1          10-19 years   0.012   0.046    0.033    0.052    0.034
2          20-29 years   0.075   0.153    0.128    0.168    0.181
3          30-39 years   0.100   0.162    0.136    0.163    0.148
4          40-49 years   0.131   0.181    0.186    0.165    0.173
5          50-59 years   0.175   0.194    0.190    0.194    0.181
6          60-69 years   0.133   0.113    0.123    0.099    0.135
7          70-79 years   0.130   0.076    0.094    0.055    0.034
8   80 years and older   0.239   0.062    0.097    0.081    0.055

**Data for gender**

Now we want to collect gender distribution in same type of bar plot for the Scandinavian coutries. The following dataframe is is used which has similiar form as for the other data that is collected for gender of the other countries. We remove the row of total number of cases and normalize data for each gender. 

In [0]:
sw_gender_bar = sw_gender
sw_gender_bar.columns = ['Age', 'Sweden']
#Drop total number of cases
sw_gender_bar = sw_gender_bar[1:3]
sw_gender_bar

Age  Sweden
1  Women    9570
2    Men    7997

In [0]:
nw_gender = nw_gender.copy()
nw_gender_bar = sw_gender_bar.copy()
nw_women = [int(i) for i in nw_gender['Women']]
nw_men = [int(i) for i in nw_gender['Men']]
nw_gender_bar['Norway'] = [sum(nw_women), sum(nw_men)] 
nw_gender_bar = nw_gender_bar
nw_gender_bar = nw_gender_bar.drop(columns = ['Sweden'])
#nw_gender_bar
#########################
fn_gender_bar = fn_gender
fn_gender_bar = pd.DataFrame(fn_gender_bar['Gender,NumberOfCases'].str.split(',', 1).tolist(), 
                             columns = ['Age','Finland'])
fn_gender_bar.loc[-1] = fn_gender_bar.loc[0]
fn_gender_bar.loc[-2] = fn_gender_bar.loc[1]
fn_gender_bar = fn_gender_bar.drop(fn_gender_bar.index[0:2])
fn_gender_bar.index = [0, 1, 2]
fn_gender_bar = fn_gender_bar[1:3]
##fn_gender_bar 
################################
dk_gender = dk_age_gender.drop(columns =['Age'])
dk_data = pd.DataFrame({'Denmark':[dk_gender['NumberOfCases'].sum(), dk_gender['Women'].sum(), dk_gender['Men'].sum()]})
dk_gender_bar = fn_gender_bar
dk_gender_bar['Denmark'] = dk_data
dk_gender_bar = dk_gender_bar.drop(columns = ['Finland'])
#dk_gender_bar

###########################
ic_gender_bar = ic_gender
ic_gender_bar.columns = ['Age', 'Iceland']
ic_gender_bar = ic_gender_bar[1:3]
#ic_gender_bar

In [0]:
#collect the code in same dataframe for every countries: 
bar_stack_gender = sw_gender_bar.copy()
bar_stack_gender['Sweden'] = bar_stack_gender['Sweden']/bar_stack_gender['Sweden'].sum()
bar_stack_gender['Norway'] = nw_gender_bar['Norway']/nw_gender_bar['Norway'].sum()
bar_stack_gender['Denmark'] = dk_gender_bar['Denmark']
bar_stack_gender['Denmark'] = bar_stack_gender['Denmark']/bar_stack_gender['Denmark'].sum()
bar_stack_gender['Finland'] = [int(i) for i in fn_gender_bar['Finland']]
bar_stack_gender['Finland']  = bar_stack_gender['Finland']/bar_stack_gender['Finland'] .sum()
bar_stack_gender['Iceland'] = ic_gender_bar['Iceland']
bar_stack_gender['Iceland'] = bar_stack_gender['Iceland']/bar_stack_gender['Iceland'].sum()
bar_stack_gender = bar_stack_gender.round(3)
bar_stack_gender

Age  Sweden  Norway  Denmark  Finland  Iceland
1  Women   0.545   0.502    0.574    0.524    0.504
2    Men   0.455   0.498    0.426    0.476    0.496

**Discussion of dataset statistics, including key points from exploratory data analysis.**  
From our exploratory data analysis, we were able to determine some interesting trends within the scandinavian data set. For one, we were able to see trends in the age dataset in which there appeared to be normally distributed. See data analysis for more information and analysis.

#**3. Data Analysis**

**Describe your data analysis and explain what you've learned about the dataset.**    
Much of the data analysis revolves around exploratory visual analysis. In other words, we were able to describe and speculate on the nature of the patterns in virus spread due to a series of visualizations.    





## Visualisation 1
> Impact of Covid-19 in Scandinavia 





**Total COVID-19 Cases Per Country**
 
The growth of the coronavirus in the five Scandinavian countries can be seen below: The following contains Confirmed Cases, Deaths, Recovered Cases, and active Cases for countries. By scrolling over each country line we can see country statistics, which will show the number of confirmed cases, deaths, recovery, and active cases with respect to time.


In [0]:
#import data

data=pd.read_csv("https://raw.githubusercontent.com/eng22/misc/master/COVID-19-time-series.csv")
data.tail()
data.drop(data[data['Province/State']=='Faroe Islands'].index, inplace=True)
data.drop(data[data['Province/State']=='Greenland'].index, inplace=True)

from plotly.subplots import make_subplots
fig = make_subplots( rows=2, cols=2, vertical_spacing=0.1,
    subplot_titles=("Confirmed", "Deaths", "Recovered", "Active Cases"))

dk= data.loc[data['Country/Region']=='Denmark']
fn=(data.loc[data['Country/Region']=='Finland'])
ic=(data.loc[data['Country/Region']=='Iceland'])
sw=(data.loc[data['Country/Region']=='Sweden'])
nw=(data.loc[data['Country/Region']=='Norway'])

In [0]:


#Confirmed Cases: Plot 1
fig.add_scatter(x=dk['Date'],y=dk['Confirmed'],
                name='Denmark', text='Denmark', row=1, col=1,
                marker={'symbol':'line-ew', 'color': 'red'})
fig.add_scatter(x=fn['Date'], y=fn['Confirmed'],
                name='Finland', text='Finland', row=1, col=1,
                marker={'symbol':'line-ew', 'color': 'blue'})
fig.add_scatter(x=ic['Date'], y=ic['Confirmed'],
                name='Iceland', text='Iceland', row=1, col=1,
                marker={'symbol':'line-ew', 'color': 'green'})
fig.add_scatter(x=nw['Date'],y=nw['Confirmed'],
                name='Norway', text='Norway', row=1, col=1,
                marker={'symbol':'line-ew', 'color': 'orange'})
fig.add_scatter(x=sw['Date'], y=sw['Confirmed'],
                name='Sweden', text='Sweden', row=1, col=1,
                marker={'symbol':'line-ew', 'color': 'LightSkyBlue'})

#Deaths: Plot 2
fig.add_scatter(x=dk['Date'], y=dk['Deaths'],
                name='Denmark', text='Denmark',row=1, col=2, 
                marker={'symbol':'line-ew', 'color': 'red'})
fig.add_scatter(x=fn['Date'], y=fn['Deaths'],
                name='Finland', text='Finland', row=1, col=2,
                marker={'symbol':'line-ew', 'color': 'blue'})
fig.add_scatter(x=ic['Date'],y=ic['Deaths'],
                name='Iceland', text='Iceland', row=1, col=2,
                marker={'symbol':'line-ew', 'color': 'green'})
fig.add_scatter(x=nw['Date'],y=nw['Deaths'],
                name='Norway', text='Norway', row=1, col=2,
                marker={'symbol':'line-ew', 'color': 'orange'})
fig.add_scatter(x=sw['Date'], y=sw['Deaths'],
                name='Sweden', text='Sweden', row=1, col=2,
                marker={'symbol':'line-ew', 'color': 'LightSkyBlue'})

#Recovered Cases: Plot 3
fig.add_scatter(x=dk['Date'],y=dk['Recovered'],
                name='Denmark', text='Denmark', row=2, col=1,
                marker={'symbol':'line-ew', 'color': 'red'})
fig.add_scatter(x=fn['Date'],y=fn['Recovered'],
                name='Finland', text='Finland', row=2, col=1,
                marker={'symbol':'line-ew', 'color': 'blue'})
fig.add_scatter(x=ic['Date'],y=ic['Recovered'],
                name='Iceland', text='Iceland', row=2, col=1,
                marker={'symbol':'line-ew', 'color': 'green'})
fig.add_scatter(x=nw['Date'],y=nw['Recovered'],
                name='Norway', text='Norway', row=2, col=1,
                marker={'symbol':'line-ew', 'color': 'orange'})
fig.add_scatter(x=sw['Date'],y=sw['Recovered'],
                name='Sweden', text='Sweden', row=2, col=1,
                marker={'symbol':'line-ew', 'color': 'LightSkyBlue'})


#Active Cases: Plot 4
fig.add_scatter(x=dk['Date'],
                y=dk['Active'],
                name='Denmark', text='Denmark', row=2, col=2, 
                marker={'symbol':'line-ew', 'color': 'red'})
fig.add_scatter(x=fn['Date'],
                y=fn['Active'],
                name='Finland', text='Finland', row=2, col=2,
                marker={'symbol':'line-ew', 'color': 'blue'})
fig.add_scatter(x=ic['Date'],
                y=ic['Active'],
                name='Iceland', text='Iceland', row=2, col=2,
                marker={'symbol':'line-ew', 'color': 'green'})
fig.add_scatter(x=nw['Date'],
                y=nw['Active'],
                name='Norway', text='Norway', row=2, col=2,
                marker={'symbol':'line-ew', 'color': 'orange'})
fig.add_scatter(x=sw['Date'], y=sw['Active'],
                name='Sweden', text='Sweden', row=2, col=2,
                marker={'symbol':'line-ew', 'color': 'LightSkyBlue'})

fig.update_layout(height=900, showlegend=False, title='COVID-19 Growth in Scandinavia')

fig.show()



From the above chart, It can be observed that **Sweden's** growth is drastically increased with *deaths* and *confirmed cases* and also has the most number of active cases which is high in comparison to other similar countries. This could be due to the nature of **Sweden's** pandemic response(refer task 1), which was in comparison to other similar countries.
In contrast, with the immediate lockdown effect, **Denmark** has the highest number of *recovery* and flatten the curve for *death* and *confirmed cases*.
Since, **Finland and Norway** where the trend is pretty much similar to the spread of the virus with the immediate lockdown just within a week after Denmark has announced. However, **Norway** have the least number of *recovery* followed by **Iceland**. Considering the *Deaths* into account **Sweden** has thrice the number of death in comparison with other countries. Moreover, the less populated country **Iceland** has a steady liner growth in all, which also has the least *cases* and *number of deaths* in this pandemic. Overall, after imposing the necessary majors taken by all the Scandinavian countries except Sweden, they contained the growth of the deadly contagious disease. 



## Visualisation 2
> Infected population by age group and gender 

Here, our take is to identify and understand the influence of coronavirus on different age groups and biological sex. The below demographics depicts the number of people infected over time by age group. Flot the cursor on the plots to get a detailed insight for each county.



In [0]:
import plotly.graph_objects as go
from bokeh.palettes import Blues8, Reds

Reds = ['#a50f15', '#de2d26', '#fb6a4a', '#fc9272', '#fcbba1', '#fee5d9'] #THIS IS NEW

bar1 = go.Bar(x = bar_stack['Age'], y = bar_stack['Sweden'], 
              text = bar_stack['Sweden'], textposition='outside', name='Sweden', marker_color= Reds[0])  #And RED[i] NEW 
bar2 = go.Bar(x = bar_stack['Age'], y = bar_stack['Finland'], 
              text = bar_stack['Finland'], textposition='outside', name='Finland', marker_color= Reds[1])
bar3 = go.Bar(x = bar_stack['Age'], y = bar_stack['Norway'], 
              text = bar_stack['Norway'], textposition='outside', name='Norway', marker_color= Reds[2])
bar4 = go.Bar(x = bar_stack['Age'], y = bar_stack['Denmark'], 
              text = bar_stack['Denmark'], textposition='outside', name='Denmark', marker_color= Reds[3])
bar5 = go.Bar(x = bar_stack['Age'], y = bar_stack['Iceland'], 
              text = bar_stack['Iceland'], textposition='outside', name='Iceland', marker_color= Reds[4])
fig = go.Figure([bar1, bar2, bar3, bar4, bar5])

fig.update_layout(barmode='stack', title = 'Distribution of confirmed cases of Covid 19 by age')  #THIS IS NEW
fig.show()

The figure for the age distribution of the confirmed cases shows that most confirmed cases among every country are of age '50-59 years'. The percentage is between 17.5% to 19.4% which means the difference in percentage is at most roughly 2% between every country. Also the age group '40-49 years' are close to the percentage of '50-59 years'. It is possible to see that the data for '40-49 years' and '50-59 years' together take up around 30%-35% of the total number of cases. However, Sweden seems to have less percentage than the other countries, but a higher percentage of '70-79 years' and '80 years and older'. 

The age group '9 years and younger' shows very tiny percentages, with 'Iceland' that has the highest percentage with around 6%. The data is as discussed in part 2, classified differently, meaning there is some variance in the data. '10-19 years' has also a small percentage that is at a maximum 5.2%. '60-60 years' data does not vary a lot between the countries, except Finland that has roughly 1 % of the cases. The rest of the countries har between 10% and 13.5% of the total number of confirmed cases.

It can be observed that infants and younger people likely to get less infected. However, the middle age group is at a higher risk of getting infected.


In [0]:
import plotly.graph_objects as go
from bokeh.palettes import Blues8,RdGy


color = '#404040', '#bababa', '#ffffff', '#f4a582', '#ca0020'

bar1 = go.Bar(x = bar_stack_gender['Age'], y = bar_stack_gender['Sweden'], 
              text = bar_stack_gender['Sweden'], textposition='outside', name='Sweden', marker_color= color [0])
bar2 = go.Bar(x = bar_stack_gender['Age'], y = bar_stack_gender['Finland'], 
              text = bar_stack_gender['Finland'], textposition='outside', name='Finland', marker_color= color [1])
bar3 = go.Bar(x = bar_stack_gender['Age'], y = bar_stack_gender['Norway'], 
              text = bar_stack_gender['Norway'], textposition='outside', name='Norway', marker_color= color [2])
bar4 = go.Bar(x = bar_stack_gender['Age'], y = bar_stack_gender['Denmark'], 
              text = bar_stack_gender['Denmark'], textposition='outside', name='Denmark', marker_color= color [3])
bar5 = go.Bar(x = bar_stack_gender['Age'], y = bar_stack_gender['Iceland'], 
              text = bar_stack_gender['Iceland'], textposition='outside', name='Iceland', marker_color= color [4])
fig = go.Figure([bar1, bar2, bar3, bar4, bar5])

fig.update_layout(barmode='stack', 
                  title="Distribution of confirmed cases by gender of Covid 19")
fig.show()

When looking at the distribution of the confirmed cases by gender, it easy to observe that the data is almost 50/50 which indicates that genders does not seem to play significant role on the effect of Covid-19. Although the difference is small, every country has most cases of women. However, this can be a result of the phycological difference between the genders. More women are worried about their health care situation than men, according to a Bankrate.com survey. This means that women might have lower threshold for testing themselves based on flu symptoms.


## Visualisation 3
> Population density : cases per thousand  

We take a look at the different trends of virus growth in each country. Further want to see an indication of other trends upholding factors such as population density. The below illustration shows the map of Scandinavia. You can get an overview of the map which shows population density by confirmed cases per 1000 for each country with a darker or lighter shade.





In [0]:
def popdens(row):
  if row['Code'] == 'DNK':
    return row['5/13/20'] / pop.at[0, 'Population'] * 1000
  if row['Code'] == 'FIN':
    return row['5/13/20'] / pop.at[1, 'Population'] * 1000
  if row['Code'] == 'ISL':
    return row['5/13/20'] / pop.at[2, 'Population'] * 1000
  if row['Code'] == 'NOR':
    return row['5/13/20'] / pop.at[3, 'Population'] * 1000
  if row['Code'] == 'SWE':
    return row['5/13/20'] / pop.at[4, 'Population'] * 1000
  else:
    return "error"

In [0]:
pop = pd.read_csv('https://raw.githubusercontent.com/eng22/misc/master/Populations.csv')
data = pd.read_csv('https://raw.githubusercontent.com/CSSEGISandData/COVID-19/master/csse_covid_19_data/csse_covid_19_time_series/time_series_covid19_confirmed_global.csv')
my_countries = ['Denmark', 'Finland', 'Iceland', 'Norway', 'Sweden']
my_data = data.loc[data['Country/Region'].isin(my_countries)]
#Group the data by the scandinavian country 
my_data = my_data.groupby('Country/Region').sum()
my_data['Code'] = ['DNK', 'FIN', 'ISL', 'NOR', 'SWE' ]
my_data['CasesPerThou'] = my_data.apply (lambda row: popdens(row), axis=1)

fig = go.Figure(data=go.Choropleth(
    locations = my_data['Code'],
    locationmode='ISO-3',
    z = my_data['CasesPerThou'],
    text = my_data.loc[:, '5/13/20'],
    colorscale = 'darkmint',
    autocolorscale=True,
    reversescale=True,
    marker_line_color='darkgray',
    marker_line_width=0.004,
    colorbar_tickprefix = '',
    colorbar_title = 'Cases per 1000',
))

fig.update_layout(
    title_text='Cases per Thousand',
    geo=dict(
        scope='europe', showframe=False,
        showcoastlines=False,
        projection_type='equirectangular'
    ),
    annotations = [dict(
        x=0.55,
        y=0.1,
        xref='paper',
        yref='paper',
        text='',
        showarrow = False
    )]
)

fig.show()

Looking at the map and hovering the cursor will show the number of COVID-19 cases with respect to the population of each country. Here, the lighter shade addresses the less number of confirmed cases per 1000 occupants of the country. Where Finland has the lowest ratio of 1.07 followed by Norway and Denmark, 1.48 and 1.88 respectively. In contrast, Sweden is the second highest at 2.75. Interestingly, Iceland has the highest percentage, but the lowest number of total cases.


## Visualisation 4
> Test performend by diffenent countries to contain the growth  

Below, we see the testing results from each country. The "percent positive" is the percent of tests that turned out to be positive, out of every test conducted.  

Note that each line begins when the country begins to record/release their testing statistics.





In [0]:
def test_ratio(row):
  if row['total_tests']==0:
    return 0
  if row['total_cases'] / row['total_tests'] == 1:
    return 0
  return row['total_cases'] / row['total_tests']

In [0]:
#population data
from plotly.subplots import make_subplots
pop = pd.read_csv('https://raw.githubusercontent.com/eng22/misc/master/Populations.csv')
#data.plot(y=['total_cases', 'total_tests'], use_index=True, figsize=(10,10))

test = pd.read_csv('https://raw.githubusercontent.com/eng22/misc/master/owo.csv')

#Create percentage of positive tests in dataframe
test['Percent Positive Tests'] = test.apply (lambda row: test_ratio(row), axis=1)

fig = make_subplots(rows=1, cols=1)


#fig = px.line(test.loc[test['iso_code']=='DNK'], x='date', y='Percent Positive Tests')


fig.add_scatter(x=(test.loc[test['iso_code']=='DNK'])['date'],
                y=(test.loc[test['iso_code']=='DNK'])['Percent Positive Tests'],
                name='Denmark', text='Denmark')

fig.add_scatter(x=(test.loc[test['iso_code']=='NOR'])['date'],
                y=(test.loc[test['iso_code']=='NOR'])['Percent Positive Tests'],
                name='Norway', text='Norway')

fig.add_scatter(x=(test.loc[test['iso_code']=='ISL'])['date'],
                y=(test.loc[test['iso_code']=='ISL'])['Percent Positive Tests'],
                name='Iceland', text='Iceland')


fig.add_scatter(x=(test.loc[test['iso_code']=='SWE'])['date'],
                y=(test.loc[test['iso_code']=='SWE'])['Percent Positive Tests'],
                name='Sweden', text='Sweden')

fig.add_scatter(x=(test.loc[test['iso_code']=='FIN'])['date'],
                y=(test.loc[test['iso_code']=='FIN'])['Percent Positive Tests'],
                name='Finland', text='Finland')

fig.update_layout(
    title_text = 'Percent of Positive COVID-19 Tests'
)


fig.show()

The above-annotated graph depicts that all the countries began testing for suspects from the 25 Feb 2020 after imposing the national lockdown. It is observed that Sweden has a different strategy to contain the virus as they started tests aggressively with testing more people in comparison to other countries since they have not imposed a lockdown after identifying the first case of coronavirus. However, rest of the countries continued the testing linearly after two weeks where the growth of the virus is pretty much contained (refer Visualisation 1) 

**If relevant, talk about your machine-learning.**

Throughout the course of this project, the emphasis has been moved to data analysis and interpretation of anomalies rather than pattern forecasts, so we do not use machine learning models in this project.

#**4. Genre**
**Which genre of data story was used?**  
The seven basic genres in narrative visualization are magazine-style, annotated chart, partitioned poster, flow chart, comic strip, slide show, and film/video/animation. 

Such genre varies mainly from the point of view of:


> Distinct visual scenes, multiplexed in time and/or space.

> The ordering of their visual elements.




However, these categories are not generally exclusive: they may work in the same way as key components, merging to create more diverse visual styles. All these genres perform for a specific task for the specific narrate type.  Picking the appropriate genre rely on several factors, including the type of story, type of dataset, target audiences, the sophistication of the plot. 

We considered ***'Annotated Graphs/maps'*** to address the visualization since we are showcasing the comparison of the response to COVID-19 between Scandinavian countries, it will be convenient to outcast the features like multi-messaging where it can communicate clearly through the interaction of the text with the annotations and graphic elements. Users can discover additional details as per the requirements by hovering highlights and use other tools. This also retains a consistent visual platform that only alters content within each panel while leaving the general layout of the visual elements unchanged. There are several other genres that has been covered along with the annotated graphs.[2]


**Which tools did you use from each of the 3 categories of Visual Narrative (Figure 7 in Segal and Heer). Why?**

Among the three design space dimensions, visual narrative helps and strengthens the visualization. It is recognized that some unavoidable rationality depends upon the application of categorization on a complex collection of designs. In the period of analysis, **Visual Structuring & Highlighting** tools are used from the visual narrative tactics.

*   **Visual structuring** refers to convey the overall structure of the Covid-19 impact on the viewer and allow him to define his role in the wider establishment of the depiction. These visual techniques serve to direct the viewer in advance so that they can track his success through the annotated graphs and map in visualization.

*   **Highlighting contributes** to visual narrative tactics by assisting to draw the viewer's attention to specific elements in the show. This is accomplished by using a different color for line charts and histograms, picture, and the size of the illustration. This would increase the relevance of the item relative to its surrounding elements.[2]


**Which tools did you use from each of the 3 categories of Narrative Structure (Figure 7 in Segal and Heer). Why?**

A further dimension of space design that describes is 'Narrative Structure' which referred by visualization and non-visualization structure to support and promote the narrative. **Interactivity & Messaging** are the two tools have been used to address the Narrative tactics in the project.

* **Interactivity** focuses on how the user can control the visualization (filtering, picking, scanning, and navigating) and even how the user understands the visualization. Interactivity is performed in the study based on the custom tools that the user can take advantage of to visualize the comparison between different countries by selecting different countries to check the status against the number of infected cases, deaths recovery in the specific graph.

*	**Messaging** addresses the method of visualization to convey insights and suggestions to the viewer. This is achieved through adding short text fields, labels & captions to the plots and headlines for the illustrations. It is more substantial descriptions of introductions summaries.[2]



#**5. Visualizations**
**Explain the visualizations you've chosen.
Why are they right for the story you want to tell?**

With our narrative of data visualization, we made the following illustrations to understand and address the analysis of the effect and response of Scandinavia to the Coronavirus.

1. **Impact of Covid-19 in Scandinavia** 

>  A line chart category of the annotated genre is used to visualize the impact of COVID 19 in Scandinavian countries. The plot is divided into 4 parts, 'Confirmed case', 'Deaths', 'Recovery', and 'Active cases'. The user can notice the trend about the growth of the virus over time. This chart also has the feature of mouse-hover and multi-color distribution for different countries with the help of this the user and get details as per convenience about the number of who get infected, died, currently in the state of recovery, and fully recovered on a particular date and for a particular country

2. **Infected population by age group and gender**

> The goal for this visualization is we wanted to analyze the age group and the biological sex who has suffered by the virus. Our choice for this visualization is a bar plot as it was the right option to justify the analysis of the number of infected people. Bar graphs are painless to understand and can be used to show changes over time. As we wanted to showcase and compare the number of confirmed cases with the different categories of age group and gender, and this achieved successfully with the bar plots. We have also added the feature of hovering using different color shades for the countries.

3. **Population Density**

> The population density with the number of confirmed cases per 1000 is depicted by using choropleth maps. Choropleth maps are also a form of an annotated genre where we have displayed the map of the Scandinavian countries. The user can get an overview of the map which shows population density for each country with a darker or lighter shade. One of the reasons for using this annotation is it benefits the user to understand and get insights about spatial distributions of data quite well.  We have distributed the data by the infected patient per thousand concerning the population of the particular country.

4. **Test performed by different countries**

> We also wanted to analyze the growth of the virus over time after imposing the restriction and performing the test on suspected people to check whether the growth is contained or not. And to get visual narrative we used the line chart by distributing it with the test performed with respect to time to understand the trend. We have used a line graph here to address change over the short and long term as it shows even small changes over time. To ease the understanding of the user we also used a set of colors and details on-demand features.


Overall, after imposing national lockdown with the immediate effect of virus spread, all the Scandinavian countries contained the virus growth and flatter the curve except Sweden. As Sweden has taken another route to tackle the situation by not imposing lockdown and testing the positive patients aggressively.

#**6. Discussion**
**What went well?**  
We showcased the impact of Covid-19 globally though the annotated maps and visual structuring the case death and recovery. With our resources and available data, we were able to analyze the influence of the virus spread by illustrating the comparison between the Scandinavian countries by population density with respect to infected cases. We were also able to identify the most suffered age group and gender in the outbreak.
 
The statistic gives the documentation about impact of the virus spread in regard to the population density of Scandinavian countries with the help of geo maps. Comparison between the Scandinavian and their responses to covid 19 gives insights on the trend about the testing of the infected person which is performed over time to contain the virus.

**What is still missing/could be improved?**  
One direction that this project could take could be the prediction of future growth. Possibly, we could use our knowledge of machine learning in order to predict each country's future action to protect the people who are more at risk. 

Additionally, our further study can be enhanced by including the following analysis:
> A analysis of the "country of origin" data for each infection instance.

> Address the response of the population to the country lockdown.

> Showcasing the current availiblity and Predicting the rescources of medical needs and hospital care resources, performed with the advance machine learning which might need the large data set.  

#**7. Contributions**
The Final project is executed by starting with the traditional method of brain stroming about finalising the topic, accumulation of data collection and performing visualisations. However, each member of the team contributed and assignend task to the project equally. Furthermore, the following addresses the task responsibility of each team member with the mentioned task from table of content.

> **Emily Grigg** *(s196492)* 

* **Assignment A**

^ Voiceover

^ Data Stasticts 

* **Assignment B**

^ Website archtiecture and modeling.

^ Task 3 (Visualisation 1, 3, 4), Task 5, Task 6 

^ Analysis (Visualisation 1, 3, 4)

^^ Explainer Notebook Structuring & Moderation


> **Shubham Ingole** *(s200092)*

* **Assignment A** 

^ Video Structuring

^ Designing

* **Assignment B**

^ Data Collection 

^ Website content

^ Narration of Data story

^ Explainer Notebook Structuring & Moderation

^ Github Setup

^ Task 1 ,Task 2, Task 3(Visualisation 1), Task 4,Task 5,Task 6

> **Thakshina Tharmapalan** *(s196628)*

* **Assignment A** 

^ Data collection

^ Rendering

* **Assignment B**

^ Data Manipulation

^ Data Cleaning

^ Task 2, 3 (Visualisation 2&3), Task 5, Task 6 

^ Analysis (Visualisation 2&3)

^ Explainer Notebook Structuring & Moderation



#**References**

[1] [World health orginisation](https://www.who.int/)

[2] Edward Segel and Jeffrey Heer, "Narrative Visualization: Telling Stories with Data".

[3] Pia Hansson. ["The Nordics and COVID-19: Together or Apart?"](https://www.diis.dk/en/research/the-nordics-and-covid-19-together-or-apart) 